NACT/attention_query_utilities/attention_query.py

```diff
@@ -441,85 +441,89 @@ def set_scanpy_object(self, new_data):
+
+        Raises:
+        None
         """
 
-        if top_n_genestrings_dict is None:
-            try:
-                att_dict = self.top_n_genestrings_dict.copy()
-                print("==> Since no dictionary was provided, we will use gene"
-                      " names dictionary as default")
-            except Exception as _:
-                print("==> Please either provide a top n gene name list, or set"
-                      "the attribute 'self.top_n_genestrings_dict'")
-        else:
-            att_dict = top_n_genestrings_dict
-
-        top_n_genes = pd.DataFrame.from_dict(att_dict)
-
-        def tf(genes):
-            """Calculate the gene frequency of a list of genes
-            ***
-            @Oscar: This needs to be cleaned up a bit and maybe moved to a
-            different scope : )
-            ***
-            """
-            freq = {}
-            total = len(genes)
-
-            # count occurrences
-            for gene in genes:
-                if gene in freq:
-                    freq[gene] += 1
-                else:
-                    freq[gene] = 1
-
-            # generate frequencies
-            for gene in freq:
-                freq[gene] = freq[gene] / total
-            return freq
-
-        def idf(dictionaries):
-            """Calculate the inverse document frequency of genes in a list of
-               dictionaries.
-
-            ***
-            @Oscar: This needs to be cleaned up a bit and maybe moved to a
-            different scope : )
-            ***
-            """
-            freq = {}
-            for document in dictionaries:
-                for gene in document:
-                    if gene in freq:
-                        freq[gene] += 1
-                    else:
-                        freq[gene] = 1
-            for gene in freq:
-                freq[gene] = np.log(len(dictionaries) / freq[gene])
-            return freq
-
-        tf_idf = {}
-        # calculate gene frequencies
-        print("==> Calculating gene frequencies")
-        tf_vals = top_n_genes.apply(tf, axis=1)
-
-        # calculate inverse document frequencies
-        print("==> Calculating inverse document frequencies")
-        idf_vals = idf(tf_vals)
-
-        # calculate tf-idf values for all genes in ranked_topN_strdf
-        print("==> Calculating tf-idf values for all ranked genes")
-        for document in range(len(tf_vals)):
-            for gene in tf_vals[document]:
-                tf_idf[gene] = tf_vals[document][gene] * idf_vals[gene]
-        return tf_idf
+
+        # Convert the intial inputs to dataframes 
+        topgenes_df = pd.DataFrame.from_dict(top_n_names) # dataframe containing gene names sorted by attention sums
+        attention_sums = pd.DataFrame.from_dict(top_n_df_dict) # dataframe containing attention sums
+
+
+        # Check if initial input is a dataframe
```

Comment:
**L**: This will not check the input data but rather the variable declared above. If there is an issue with the input not being a `dict`, this will not catch it (based on the comment here)